README.md

```diff
@@ -131,7 +131,314 @@ The Docker image will be tagged with:
+- A custom Docker network for container communication
+
+### Environment Configurations
+
+Each environment has dedicated containers with specific ports:
+
+- **Production**
+  - agents-prod (ports: 5010, 3010)
+  - prosper-prod (ports: 5020, 3020)
+- **Staging**
+  - agents-staging (ports: 5030, 3030) 
+  - prosper-staging (ports: 5040, 3040)
+- **Development**
+  - agents-dev (ports: 5050, 3050)
+  - prosper-dev (ports: 5060, 3060)
+
+### Deployment Process
+
+The deployment process is automated through GitHub Actions workflows:
+
+1. **Production Deployments** (`docker-prod.yml`)
+   - Triggered by publishing a new release
+   - Tags images with semantic version numbers (e.g., v1.2.3, 1.2, 1)
+   - Deploys to production environment
+
+2. **Staging Deployments** (`docker-staging.yml`)
+   - Triggered by pushes to the `main` branch
+   - Tags images with commit SHA and 'staging' tag
+   - Deploys to staging environment
+
+3. **Development Deployments** (`docker-dev.yml`)
+   - Triggered by pushes to non-main branches
+   - Tags images with commit SHA and 'dev' tag
+   - Deploys to development environment
+
+### Deployment Script
+
+The `deploy.sh` script handles the actual deployment process:
+
+1. Creates/ensures a Docker network exists
+2. Sets up and manages a PostgreSQL container
+3. Creates separate databases for each service
+4. Pulls the latest Docker images
+5. Deploys or updates containers with appropriate configurations
+6. Manages environment variables and port mappings
+
+### Database Management
+
+- Each service gets its own database within a shared PostgreSQL instance
+- Database names are automatically generated based on container names
+- Data persistence is handled through a Docker volume
+
+### Version Management
+
+- Production releases use semantic versioning (v1.2.3)
+- Staging builds are tagged with commit SHAs and 'staging'
+- Development builds are tagged with commit SHAs and 'dev'
+- All images are stored in GitHub Container Registry (ghcr.io)
+
+### Automatic Updates
+
+The deployment script checks for image updates before deploying:
+- Compares current and new image IDs
+- Only deploys if there's a new version available
+- Maintains container state and configurations
+
+## Deployment to TEE environment
+
+Reality Spiral can be deployed to a Trusted Execution Environment (TEE) using the [Oasis SDK](https://github.com/oasisprotocol/oasis-sdk). Below we are describing how to deploy Reality Spiral to a TEE environment using the Oasis ROFL tool.
+
+### Prerequisites
+
+Install the Oasis CLI using the latest release from the [Oasis CLI releases page](https://github.com/oasisprotocol/cli/releases).
+
+### Account Setup
+
+Create a new account using the Oasis CLI.
+
+```sh
+oasis-cli account create --name <account-name>
+```
+
+Request testnet tokens from the [Oasis Faucet](https://faucet.testnet.oasis.dev/).
+
+### Initialize and create the ROFL application
+
+Initialize the ROFL application.
+
+```sh
+oasis rofl init
```

Comment:
Calling `oasis rofl init` should be omitted, because `rofl.yaml` is already part of the repository.